tests/services/phase.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+      });
+    });
+
+    describe('.updateSessionParameters()', () => {
+      it('should update the currentPhase and repoName based on given sessionData', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+
+        expect(phaseService.currentPhase).toEqual(Phase.phaseBugReporting);
+        expect(phaseService.repoName).toEqual(mockSettingsFile[Phase.phaseBugReporting]);
+      });
+    });
+
+    describe('.attemptSessionAvailabilityFix()', () => {
+      it('should throw no errors given phaseBugReporting and student role', () => {
+        userService.currentUser = testStudent;
+        // refresh phaseService with new userService
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).not.toThrowError();
+      });
+
+      it('should throw an error given no openPhases', () => {
+        userService.currentUser = testStudent;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Current Phase\'s Repository has not been opened.')
+        );
+      });
+
+      it('should throw an error given non-student role and phaseBugReporting', () => {
+        userService.currentUser = testTutor;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Bug-Reporting Phase\'s repository initialisation is only available to Students.')
+        );
+      });
+    });
+
+    describe('.githubRepoPermissionLevel()', () => {
+      it('should return "repo" if phaseModeration is included in openPhases', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.sessionData = {
+          ...mockSessionData,
+          openPhases: [ Phase.phaseModeration ]
+        };
+        expect(phaseService.sessionData.openPhases).toContain(Phase.phaseModeration);
+        expect(phaseService.githubRepoPermissionLevel()).toEqual('repo');
+      });
+
+      it('should return "public_repo" if phaseModeration is not included in openPhases', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.sessionData = {
+          ...mockSessionData,
+          openPhases: [ ]
```

Comment:
The space here between the braces can be removed too?

Reply:
👍